pino.d.ts

```diff
@@ -192,10 +105,98 @@ interface LoggerExtras extends EventEmitter {
 
 declare namespace pino {
     //// Exported types and interfaces
+    
+    interface BaseLogger {
```

Comment:
why move it from top level into a namespace?

Reply:
Because it never export outside.

Reply:
can't we export it instead?

Reply:
Then the old way of import is missing this exportation.
e.g.
```
import pino from 'pino'
pino.BaseLogger
```

Reply:
We generally want to move away from relying on namespace. See https://github.com/pinojs/pino/issues/1193#issuecomment-966899139
I agree that types should be fixed if we lost the way to do something we could do before, though.

Reply:
I agree that it should move away from namespace, but the current version is too late to do it.
Current version should support the import syntax that can be used in `7.0.0`.

The complete removal should be happen in next major `8.0.0`.

Reply:
I understand that we can't remove it yet, but can we at least already provide alternative for people who want to do direct named imports? That was the gist of #1193 and it seems like the right direction.
Other than that, PR LGTM.

Reply:
As I mention on line 768, it has the top-level export mapped.

Reply:
This PR can import the type I moved inside `namespace` with the below format.

```ts
// named import
import { BaseLogger } from 'pino'
// namespace import / default import
import pino from 'pino'
pino.BaseLogger
// pino named import
import { pino } from 'pino'
pino.BaseLogger
// old P named import
import { P } from 'pino'
P.BaseLogger
```

If anything is missing, please let me know.

Reply:
Thank you for clarification!